src/main/java/duckling/responders/FileContents.java

```diff
@@ -0,0 +1,56 @@
+package duckling.responders;
+
+import duckling.requests.Request;
+import duckling.ResponseHeaders;
+
+import java.io.*;
+
+import static java.net.URLConnection.guessContentTypeFromName;
+import static java.net.URLConnection.guessContentTypeFromStream;
+
+public class FileContents extends Responder {
+    private File file;
+
+    public FileContents(Request request, OutputStream outputStream) {
+        super(request, outputStream);
+
+        this.file = request.getFile();
+    }
+
+    @Override
+    public boolean matches() {
+        return this.file.exists() && !this.file.isDirectory();
+    }
+
+    @Override
+    public void respond() throws IOException {
```

Comment:
Abstract classes in general are pretty nice!  I'm really digging Java a lot so far.

If you have some time, could you look at the `RespondersTest`/`Responders` class pair and see if there's a better way to do that pattern?  My goal here is to be able to just implement a responder for the majority of use cases.

In Ruby pseudocode, I'd generally do something like this:

```ruby
module Responders
  attr_reader :request, :output_stream
  def initialize(request, output_stream)
    @request = request
    @output_stream = output_stream
  end

  def get
    candidates = [ FileContents, FolderContents, NotFound ]
    match = candidates.filter { |candidate| candidate.matches?(request) }
    match.new(request, output_stream)
  end

end
```

I had tried to do a similar thing by doing a  `Stream<Class>` with `Class` references, but that got crazy fast.